test/asn1_time_test.c

```diff
@@ -433,7 +433,11 @@ static int convert_asn1_to_time_t(int idx)
 
     testdateutc = ossl_asn1_string_to_time_t(asn1_to_utc[idx].input);
 
-    if (!TEST_time_t_eq(testdateutc, asn1_to_utc[idx].expected)) {
+    if (
+#ifdef _WIN32
+        testdateutc != asn1_to_utc[idx].expected &&
+#endif
+        !TEST_time_t_eq(testdateutc, asn1_to_utc[idx].expected)) {
```

Comment:
I am confused how this can help with any test failure. The !TEST_time_t_eq() should do exactly the same as the condition you've added.

Reply:
`TEST_time_t_eq(..., t1, t2)` calls `ASN1_TIME_set(NULL, tn)` for each n = 1, 2, and fails if any of its return value is equal to `NULL`.

https://github.com/openssl/openssl/blob/fc2be2d07acc0cfe954320c2491b8c5461cbef09/test/testutil/tests.c#L452-L469

It could be that the bug really lies within `TEST_time_t_*` or `ASN1_TIME_set`.

Reply:
I somehow missed the special implementation of the time_t comparisons. In that case I think it should be handled in the DEFINE. Or first find out why the ASN1_TIME_set returns NULL in this case.

Reply:
Hmm, thinking about it more - this probably fails in gmtime_s() call on the -1 time_t value which is probably legitimate failure. I really wonder why the time_t comparison does all the ASN1_TIME_set() machinery instead of comparing the time_t values directly.

Reply:
See my conversation with @paulidale : https://github.com/openssl/openssl/pull/4797#discussion_r961448645 So the proper fix would be to fix the DEFINE_TIME_T_COMPARISON().

Reply:
The test suite already checks that that -1 works or not, and should skip tests with a negative time_t